src/videoCache.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * This module interacts with the server used to cache video ad content to be restored later.
+ * At a high level, the expected workflow goes like this:
+ *
+ *   - Request video ads from Bidders
+ *   - Generate IDs for each valid bid, and cache the key/value pair on the server.
+ *   - Return these IDs so that publishers can use them to fetch the bids later.
+ *
+ * This trickery helps integrate with ad servers, which set character limits on request params.
+ */
+
+import { ajax } from './ajax';
+import { tryAppendQueryString } from './utils';
+
+const PUT_URL = 'https://prebid.adnxs.com/pbc/v1/put'
+const GET_URL = 'https://prebid.adnxs.com/pbc/v1/get?';
+const EMPTY_VAST_RESPONSE = '<VAST version="3.0"></VAST>'
```

Comment:
I must admit... I'm a bit confused.

Section 2.4.2.4 of [the official 3.0 spec](https://www.iab.com/wp-content/uploads/2015/06/VASTv3_0.pdf) says that an empty tag is the correct response... but you're right that the XSD doesn't call it valid, and [their official VAST validator](https://iabtechlab.com/IABVAST/#!vast) doesn't either.

Honestly not sure how to interpret that, except to say that I'll test it against the major video players and make sure they properly play the video with no ad.